src/AbstractSerializer.php

```diff
@@ -114,15 +114,8 @@ protected static function splitStream(StreamInterface $stream)
             $headers[$currentHeader][] = $value . ltrim($line);
         }
 
-        $body = new Stream('php://temp', 'wb+');
-        if (! $stream->eof()) {
-            while ($data = $stream->read(4096)) {
-                $body->write($data);
-            }
-            $body->rewind();
-        }
-
-        return [$headers, $body];
+        // use RelativeStream to avoid copying initial stream into memory
+        return [$headers, new RelativeStream($stream, $stream->tell())];
```

Comment:
One way to test this is creating a long string body and verifying that `read()` was not called more than N times (thinking aloud)